### PR TITLE
Add flow support

### PR DESCRIPTION
**Pull request by baelter**

### CODE REVIEW

File: shard.yml [hunk 1]
```diff
@@ -3,10 +3,12 @@ version: 0.3.2
 
 authors:
   - Carl Hörberg <carl@cloudamqp.com>
+  - Anders Bälter <anders@cloudamqp.com>
 
 dependencies:
   amq-protocol:
     github: cloudamqp/amq-protocol.cr
+    branch: master
```
Comment (carlhoerberg) [inline shard.yml]: tag a new version in amq-protocol instead

File: src/amqp-client/channel.cr [hunk 2]
```diff
@@ -14,6 +14,7 @@ class AMQP::Client
     @confirms = ::Channel(Frame::Basic::Ack | Frame::Basic::Nack).new(1024)
     @next_msg_ready = ::Channel(Nil).new
     @log : Logger
+    @flow = true
```
Comment (carlhoerberg) [inline src/amqp-client/channel.cr]: default true?

File: src/amqp-client/channel.cr [hunk 3]
```diff
@@ -97,15 +99,25 @@ class AMQP::Client
         case frame
         when Frame::Channel::Close then close(frame)
         when Frame::Basic::Deliver then process_deliver(frame)
-        when Frame::Basic::Return then process_return(frame)
-        when Frame::Basic::Cancel then process_cancel(frame)
+        when Frame::Basic::Return  then process_return(frame)
+        when Frame::Basic::Cancel  then process_cancel(frame)
+        when Frame::Channel::Flow  then server_flow(frame.active)
```
Comment (carlhoerberg) [inline src/amqp-client/channel.cr]: name it `process_flow` for consistency?

**Comment by carlhoerberg** [comment]
And a `channel.flow` method so that the user can throttle the server from sending msgs to the consumers in the channel.

File: src/amqp-client/channel.cr (hunk 2 above)
Comment (baelter) [reply src/amqp-client/channel.cr]: Yeah, flow active is default state. Ticked me, no flow = don't send. flow = all good

> When a new channel is opened, it is active (flow is active) ...

https://www.rabbitmq.com/amqp-0-9-1-reference.html#channel.flow

**Pull request merged**